pkg/noderesourcetopology/podprovider/podprovider.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podprovider
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	podlisterv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	k8scache "k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+
+	apiconfig "sigs.k8s.io/scheduler-plugins/apis/config"
+)
+
+func WantsDedicatedInformer(cacheConf *apiconfig.NodeResourceTopologyCache) bool {
+	if cacheConf == nil {
+		return false
+	}
+	if cacheConf.InformerMode == nil {
+		return false
+	}
+	infMode := *cacheConf.InformerMode
+	return infMode == apiconfig.CacheInformerDedicated
+}
+
+func NewFromHandle(handle framework.Handle, cacheConf *apiconfig.NodeResourceTopologyCache) (k8scache.SharedIndexInformer, podlisterv1.PodLister) {
+	dedicated := WantsDedicatedInformer(cacheConf)
+	if !dedicated {
+		podHandle := handle.SharedInformerFactory().Core().V1().Pods() // shortcut
+		return podHandle.Informer(), podHandle.Lister()
+	}
+
+	podInformer := coreinformers.NewFilteredPodInformer(handle.ClientSet(), metav1.NamespaceAll, 0, cache.Indexers{}, nil)
```

Comment:
Sorry for my ignorant but how this custom informers knows to provide pods in terminal phase as well?

Reply:
because by default the informers watch objects without filtering. However,  [the scheduler framework explicitly filters out such pods on its watcher](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L31). Thus, the scheduler side will never see pod in terminal phase if it uses the framework-provided informer.